File: web/src/api/funds.api.ts
import { get, post, http } from './http'

const fundsApi = http.copyWith({ baseURL: '/payment' })
/** 查询所有银行卡 */
export const queryPaymentList = get<Record<string, model.funds.PaymentProvider>>('/payment-list')
/** 提现渠道列表 */
export const queryWithdrawalChannels = get<model.funds.Channel[]>('/withdraw-channel-list')
/** 用户提现(新) */
export const withdraw = post<model.funds.Channel[]>('/withdraw-new')

/** 存款 */
export const deposit = fundsApi.post('/order', { loading: true })
/** 查询存款渠道 */
export const asyncConfirmDeposit = fundsApi.post('/match', { loading: true })

File: victory/lib/apis/funds.api.dart
part of 'apis.dart';

class _FundApi {
  final queryDepositChannels = http.get('/app/payment-list', decoder: _mapToList(models.VicDepositChannelsModel.fromJson));
  final createDepositOrder = http.post('/pay/payment/order', decoder: models.VicPaymentOrderModel.fromJson, loading: true);
  final queryWithdrawalChannels = http.get.list('/app/withdraw-channel-list', decoder: models.VicWithdrawalChannelModel.fromJson);
  final withdraw = http.post('/app/withdraw-new', loading: true);
  final matchTopup = http.post('/pay/payment/match', loading: true);
}
